iModelCore/GeomLibs/geom/test/PolyfaceTest/t_ClipPlaneSet.cpp

```diff
@@ -2599,6 +2599,62 @@ TEST(ClipPlaneSet,ClipOpenBoundaryOutsideClipper)
     Check::ClearGeometry("ClipPlaneSet.ClipOpenBoundaryOutsideClipper");
     }
 
+/*---------------------------------------------------------------------------------**//**
+* @bsimethod
++---------------+---------------+---------------+---------------+---------------+------*/
+TEST(ClipPlaneSet,ClipOpenBoundaryWith2Parts)
+    {
+    // curve
+    bvector<DPoint3d> poly1 {{-20,0,0},{0,-20,0},{20,0,0},{0,20,0},{0,5,0},{5,0,0},{0,-5,0},{-5,0,0},{0,5,0},{0,20,0},{-20,0,0}};
+    Check::SaveTransformed(poly1);
+    CurveVector::BoundaryType boundaryType = CurveVector::BOUNDARY_TYPE_Open;
+    CurveVectorPtr curve = CurveVector::CreateLinear(poly1, boundaryType);
+    // clipper
+    bvector<DPoint3d> poly2 {{-7,-7,0},{7,-7,0},{7,7,0},{-7,7,0},{-7,-7,0}};
+    Check::SaveTransformed(poly2);
+    ConvexClipPlaneSet convexClipPlaneSet;
+    convexClipPlaneSet.ReloadSweptConvexPolygon(poly2, DVec3d::From(0, 0, 1), 0);
+    ClipPlaneSet clipPlaneSet(convexClipPlaneSet);

```

Comment:
For future tests: here's a more straightforward construction that works for any polygon. Sweep dir is assumed to be valid if supplied; otherwise, it defaults to 001.
```suggestion
    auto clipPlaneSet = ClipPlaneSet::FromSweptPolygon(poly2.data(), poly2.size());
```

Reply:
nice. I will change this here and 7 other places!